### PR TITLE
✨handles nesting in flattenMapping

### PR DESCRIPTION
**Pull request by hlminh2000**

### CODE REVIEW

**Comment by Jephuff** [comment]
Can you paste an example mapping that this change is for?

**Comment by alex-wilmer** [comment]
back when this was the portal-graphql project I started this file https://github.com/overture-stack/arranger/blob/master/modules/mapping-utils/src/utils.test.js

It's copied over but not being used. @hlminh2000 want to take a crack setting up unit tests for mapping-utils again? I would probably prefer the tests to be split up per file.. like mapping-utils/tests which imports stuff from src. can add common fixtures etc

the whole thing is out of scope for this ticket, but it *would* in scope to make a few unit tests for this exact change which would require hooking tests back up, even quickly just to "answer jeffrey's question". you can look here for details / inspiration https://github.com/NCI-GDC/portal-graphql

**Comment by alex-wilmer** [comment]
@hlminh2000 I'm going to whip something real quick up now actually

**Comment by alex-wilmer** [comment]
@hlminh2000  https://github.com/overture-stack/arranger/pull/138

File: modules/mapping-utils/test/flattenMapping.test.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  { field: 'level_1_2.level_2_2.level_3_1', type: 'keyword' },
+  { field: 'level_1_2.level_2_2.level_3_2', type: 'keyword' },
+  { field: 'level_1_2.level_2_3', type: 'long' },
+  { field: 'level_1_2.level_2_4', type: 'keyword' },
+];
+
+test('flattenMapping', () => {
+  let actualOutput = flattenMapping(testMapping);
+  expect(actualOutput).toEqual(expectedOutput);
+});
```
Comment (hlminh2000) [inline modules/mapping-utils/test/flattenMapping.test.js]: @Jephuff hope this helps :)

**Comment by alex-wilmer** [comment]
@hlminh2000 if you rebase this with master travis will run. also you can run `npm test` or `yarn test` from the root for

**Pull request merged**